lib/private/Files/Node/Folder.php

```diff
@@ -158,7 +158,7 @@ public function newFolder($path) {
 			$fullPath = $this->getFullPath($path);
 			$nonExisting = new NonExistingFolder($this->root, $this->view, $fullPath);
 			$this->sendHooks(['preWrite', 'preCreate'], [$nonExisting]);
-			if (!$this->view->mkdir($fullPath)) {
+			if (!$this->view->is_dir($fullPath) && !$this->view->mkdir($fullPath)) {
```

Comment:
If it's a race condition, I think we should try to create the directory first and check if it exists after
```suggestion
			if (!$this->view->mkdir($fullPath) && !$this->view->is_dir($fullPath)) {
```